contracts/eosio.system/src/peer_keys.cpp

```diff
@@ -0,0 +1,62 @@
+#include <cstdint>
+#include <eosio.system/eosio.system.hpp>
+#include <eosio.system/peer_keys.hpp>
+
+#include <eosio/eosio.hpp>
+
+namespace eosiosystem {
+
+void peer_keys::regpeerkey(const name& proposer_finalizer_name, const public_key& key) {
+   require_auth(proposer_finalizer_name);
+   peer_keys_table peer_keys_table(get_self(), get_self().value);
+   check(!std::holds_alternative<eosio::webauthn_public_key>(key), "webauthn keys not allowed in regpeerkey action");
+
+   auto peers_itr = peer_keys_table.find(proposer_finalizer_name.value);
+   if (peers_itr == peer_keys_table.end()) {
+      peer_keys_table.emplace(proposer_finalizer_name, [&](auto& row) {
+         row.init_row(proposer_finalizer_name);
+         row.set_public_key(key);
+      });
+   } else {
+      const auto& prev_key = peers_itr->get_public_key();
+      check(!prev_key || *prev_key != key, "Provided key is the same as currently stored one");
+      peer_keys_table.modify(peers_itr, same_payer, [&](auto& row) {
+         row.update_row();
+         row.set_public_key(key);
+      });
+   }
+}
+
+void peer_keys::delpeerkey(const name& proposer_finalizer_name, const public_key& key) {
+   require_auth(proposer_finalizer_name);
+   peer_keys_table peer_keys_table(get_self(), get_self().value);
+
+   // not updating the version here. deleted keys will persist in the memory hashmap
```

Comment:
Is this comment still relevant?

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/185/commits/4a7f8c854fbb0edf905568a299aaa44e3819a4d2